solvable_files.sh

```diff
@@ -34,7 +34,8 @@ function correct_mtime() {
 		username=$(dirname "$username")
 	done
 
-	relative_filepath_without_username="${relative_filepath/#$username\//}"
+	first_relative_filepath_without_username="${relative_filepath/#$username\//}"
+	relative_filepath_without_username=`echo $first_relative_filepath_without_username | sed "s/\"/\\\\\\\\\"/g"`
```

Comment:
I don't get the sed command

Reply:
this is about escaping double quotes such that when we execute the sql command via command line they are not confused with the delimitors
this is to handle the extreme case where double quotes would happen inside the file name

Reply:
But why so many `\` ?

`s/\"/\\\\\\\\\"/g` would replace `"` by `\\\\"`, no?

And lets use proper naming:

`first_relative_filepath_without_username` => `relative_filepath_without_username`
`relative_filepath_without_username` => `relative_filepath_without_username_escaped`

Reply:
I can only say that without that many `\` it was not working
my understanding is that you get those three times through different shells and thus we need that many of them
I will fix the naming

Reply:
I remember testing a filename with a quote in it and it was working fine, so I am not sure it is even necessary to do that. I will retest on Monday.

Reply:
this came from https://help.nextcloud.com/t/desktop-client-3-4-0-destroys-local-time-stamp-and-keeps-uploading-data-to-server/128512/67?u=mgallien and I tested with path like this
`/var/www/html/data/john/files/l'école/test l'ecole.txt`
`/var/www/html/data/john/files/l'école/Text"File.txt`

Reply:
An error with single quotes seem more likely

Reply:
yes this is the reason to use double quotes to escape the string when calling `mysql` or `psql`

Reply:
Using base64 might be safer:

```sql
INSERT INTO active_records (password) VALUES (FROM_BASE64('$(echo -n $PASSWORD|base64)'))
```

From: https://stackoverflow.com/questions/4383135/escaping-mysql-command-lines-via-bash-scripting

Reply:
yeah why not
it took me a while to understand the idea and yes it looks like it should be less hacky

Reply:
@artonge I have applied the base64 solution